core/src/avm2/script.rs

```diff
@@ -0,0 +1,286 @@
+            .insert(method_index, method.clone());
+
+        Ok(method)
+    }
+
+    /// Load a class from the ABC file and return it's class definition.
+    pub fn load_class(
+        self,
+        class_index: u32,
+        mc: MutationContext<'gc, '_>,
+    ) -> Result<GcCell<'gc, Class<'gc>>, Error> {
+        let write = self.0.write(mc);
+        if let Some(class) = write.classes.get(&class_index) {
+            return Ok(*class);
+        }
+
+        drop(write);
+
+        let class = Class::from_abc_index(self, class_index, mc)?;
+        self.0.write(mc).classes.insert(class_index, class);
+
+        class.write(mc).load_traits(self, class_index, mc)?;
+
+        Ok(class)
+    }
+
+    /// Load a script from the ABC file and return it's script definition.
+    pub fn load_script(
+        self,
+        script_index: u32,
+        mc: MutationContext<'gc, '_>,
+    ) -> Result<GcCell<'gc, Script<'gc>>, Error> {
+        let write = self.0.write(mc);
+        if let Some(scripts) = write.scripts.get(&script_index) {
+            return Ok(*scripts);
+        }
+
+        drop(write);
+
+        let script = Script::from_abc_index(self, script_index, mc)?;
+        self.0.write(mc).scripts.insert(script_index, script);
+
+        script.write(mc).load_traits(self, script_index, mc)?;
+
+        Ok(script)
+    }
+
+    /// Load a string from the ABC's constant pool.
+    ///
+    /// This function yields an error if no such string index exists.
+    ///
+    /// This function yields `None` to signal string index zero, which callers
+    /// are free to interpret as the context demands.
+    pub fn pool_string_option(
+        self,
+        string_index: u32,
+        mc: MutationContext<'gc, '_>,
+    ) -> Result<Option<AvmString<'gc>>, Error> {
+        let mut write = self.0.write(mc);
+        if let Some(string) = write.strings.get(&string_index) {
+            return Ok(Some(*string));
+        }
+
+        if string_index == 0 {
+            return Ok(None);
+        }
+
+        let avm_string = AvmString::new(
+            mc,
+            write
+                .abc
+                .0
+                .constant_pool
+                .strings
+                .get(string_index as usize - 1)
+                .ok_or_else(|| format!("Unknown string constant {}", string_index))?,
+        );
+        write.strings.insert(string_index, avm_string);
+
+        Ok(Some(avm_string))
+    }
+
+    /// Load a string from the ABC's constant pool.
+    ///
+    /// This function yields an error if no such string index exists.
+    ///
+    /// String index 0 is always `""`. If you need to instead treat 0 as
+    /// something else, then please use `pool_string_option`.
+    pub fn pool_string(
+        self,
+        string_index: u32,
+        mc: MutationContext<'gc, '_>,
+    ) -> Result<AvmString<'gc>, Error> {
+        Ok(self
+            .pool_string_option(string_index, mc)?
+            .unwrap_or_else(|| "".into()))
```

Comment:
unwrap_or_default

Reply:
Huh. That seems to be fine now. I seem to recall `unwrap_or_default` tripping Clippy lints at some point...